src/test/java/com/exactpro/th2/sailfish/utils/ImplementationIFilterTest.java

```diff
@@ -0,0 +1,604 @@
+/*
+ *  Copyright 2021-2021 Exactpro (Exactpro Systems Limited)
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package com.exactpro.th2.sailfish.utils;
+
+import java.util.Arrays;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import com.exactpro.sf.common.messages.IMessage;
+import com.exactpro.sf.comparison.ComparatorSettings;
+import com.exactpro.sf.comparison.ComparisonResult;
+import com.exactpro.sf.comparison.MessageComparator;
+import com.exactpro.sf.configuration.suri.SailfishURI;
+import com.exactpro.sf.scriptrunner.StatusType;
+import com.exactpro.th2.common.grpc.FilterOperation;
+import com.exactpro.th2.common.grpc.Message;
+import com.exactpro.th2.common.grpc.MessageFilter;
+import com.exactpro.th2.common.grpc.RootMessageFilter;
+import com.exactpro.th2.common.grpc.SimpleList;
+import com.exactpro.th2.common.grpc.ValueFilter;
+import com.exactpro.th2.sailfish.utils.factory.DefaultMessageFactoryProxy;
+
+public class ImplementationIFilterTest extends AbstractConverterTest {
+    private static final String MESSAGE_TYPE = "TestMsg";
+    private final DefaultMessageFactoryProxy messageFactory = new DefaultMessageFactoryProxy();
+    private final SailfishURI dictionaryURI = SailfishURI.unsafeParse("test");
+    private final ProtoToIMessageConverter converter = new ProtoToIMessageConverter(
+            messageFactory, null, dictionaryURI
+    );
+
+    @Test
+    void listContainsValueFilter() {
```

Comment:
Please, try to parameterized some test for the same operation to avoid code duplication